apis/Google.Devtools.AspNet/Google.Devtools.AspNet/CloudErrorReporting/CloudErrorReportingExceptionLogger.cs

```diff
@@ -94,15 +100,32 @@ public sealed class CloudErrorReportingExceptionLogger : ExceptionLogger
         /// <inheritdoc />
         public override Task LogAsync(ExceptionLoggerContext context, CancellationToken cancellationToken)
         {
-            ReportedErrorEvent errorEvent = CreateReportRequest(context);
-            return _client.ReportErrorEventAsync(_projectResourceName, errorEvent);
+            if (_clientTask.IsCompleted)
+            {
+                ReportedErrorEvent errorEvent = CreateReportRequest(context);
+                return _clientTask.Result.ReportErrorEventAsync(_projectResourceName, errorEvent);
+            }
+
+            if (Interlocked.Increment(ref _taskCounter) <= MaxWaitingTasks)
+            {
+                return _clientTask.ContinueWith(clientTask =>
+                {
+                    if (clientTask.IsFaulted)
+                    {
+                        throw new InvalidOperationException("The client task has faulted.");
+                    }
+                    ReportedErrorEvent errorEvent = CreateReportRequest(context);
+                    return clientTask.Result.ReportErrorEventAsync(_projectResourceName, errorEvent);
```

Comment:
I don't *think* this is going to do the right thing, in terms of the return value. The task returned by `ReportErrorEventAsync` will effectively be ignored.

I'm honestly not sure what we should be doing here - `ContinueWith` seems to only want to take synchronous actions. Maybe we should just call `ReportErrorEvent` (the synchronous method) here, and use `TaskContinuationOptions.RunContinuationsAsynchronously`?